robottelo/host_helpers/contenthost_mixins.py

```diff
@@ -40,27 +40,33 @@ def OSCAP(self):
             'cbrhel': constants.OSCAP_PROFILE[f'cbrhel{self._v_major}'],
         }
 
-    @cached_property
-    def dogfood_repofile(self):
+    def dogfood_repofile(self, sat_ver=None, product=None, snap=''):
```

Comment:
Non-blocking, just a thought, we should make the default product as `satellite` here, and make `sat_ver` and `snap_ver` as kwarg?

Reply:
We rely on the host's class name like `Satellite`, `Capsule`, or `ContentHost` to distinguish the product automagically.
If we would specify a default for a product in the method header/definition, the whole conditional block in the method would be useless.
So either user explicitly specifies the product or the method will figure it out based on the host object.
I agree with converting args to kwargs to some degree, but we have only three args here. Is it worth/beneficial to use kwargs?

Reply:
Just thinking out loud here, do we going to have the satellite version installed on the satellite different from the setting object? 

If that is the case then I support the dynamic way of handling getting a satellite version. otherwise, it seems to be just overhead.

Reply:
That is a good question for @lpramuk who aims to do upgrades with robottelo in the future.

Reply:
@Gauravtalreja1 we won't be passing mysterious dicts to functions, it is upto to caller to expand the dict before calling function.
```
d = {'sat_ver': '6.12.0', 'snap_ver': '11.0' }
obj.dogfood_repofile(**d)     <- defined as dogfood_repofile(self, sat_ver=None, snap_ver=''):
```
so definitely not  this !
```
obj.dogfood_repofile(d)      <- defined as dogfood_repofile(self, ver_kwargs=None):
```